core/src/main/java/org/apache/iceberg/rest/RESTJacksonModule.java

```diff
@@ -0,0 +1,183 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.PartitionSpecParser;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.SortOrderParser;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+
+public class RESTJacksonModule {
+
+  private RESTJacksonModule() {
+  }
+
+  public static void registerAll(ObjectMapper mapper) {
+    SimpleModule module = new SimpleModule();
+    module
+        .addSerializer(TableIdentifier.class, new TableIdentifierSerializer())
+        .addDeserializer(TableIdentifier.class, new TableIdentifierDeserializer())
+        .addSerializer(Namespace.class, new NamespaceSerializer())
+        .addDeserializer(Namespace.class, new NamespaceDeserializer())
+        .addSerializer(Schema.class, new SchemaSerializer())
+        .addDeserializer(Schema.class, new SchemaDeserializer())
+        .addSerializer(PartitionSpec.class, new PartitionSpecSerializer())
+        .addDeserializer(PartitionSpec.class, new PartitionSpecDeserializer())
+        .addSerializer(SortOrder.class, new SortOrderSerializer())
+        .addDeserializer(SortOrder.class, new SortOrderDeserializer());
+    mapper.registerModule(module);
+
+    mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+    mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
```

Comment:
Can you help me understand what these configurations are doing?

Reply:
These are a workaround for Jackson since Iceberg doesn't use the standard `get/set` bean notation.  This allows Jackson to work with the fields more directly and not require custom serializers (with basically a ton of boilerplate code) for all of the request/response objects.

Reply:
would it make sense to add a comment to the code describing this workaround? I think this isn't clear to everyone stumbling across this part of the code why it's being done

Reply:
Added the comment that Dan made almost word for word, though it lives in `JsonUtil` now. Open to moving the visibility changing back into this class and changing the comment as well.